pyaedt/modules/SolveSetup.py

```diff
@@ -2585,6 +2587,76 @@ def add_eddy_current_sweep(
         self.auto_update = legacy_update
         return True
 
+    @pyaedt_function_handler()
+    def enable_control_program(self, control_program_path, control_program_args=" ", call_after_last_step=False):
+        """Enable control program option is solution setup.
+        Provide externally created executable files, or Python (*.py) scripts that are called after each time step,
+        and allow you to control the source input, circuit elements, mechanical quantities, time step,
+        and stopping criteria, based on the updated solutions.
+
+        Parameters
+        ----------
+        control_program_path : str
+            File path of control program.
+        control_program_args : str, optional
+            Arguments to pass to control program.
+            Default value is ``" "``
+        call_after_last_step : bool, optional
+            if ``True`` the control program is called after the simulation is completed.
+            Default value is ``False``.
+
+        Returns
+        -------
+        bool
+            ``True`` if successful, ``False`` if it fails.
+
+        Notes
+        -----
+        By default a control program script will be called by the pre-installed Python interpreter:
+        ``<install_path>\Win64\commonfiles\CPython\37\winx64\Release\python\python.exe``.
+        However, the user can specify a custom python interpreter to be used by setting following environment variable:
+        ``EM_CTRL_PROG_PYTHON_PATH=<path_to\python.exe>``
+
+        References
+        ----------
+        >>> oModule.EditSetup
+        """
+        if self.p_app.solution_type not in ["Transient", "TransientXY", "TransientZ"]:
+            self._app.logger.error("Control Program is only available in Maxwell 2D and 3D Transient solutions.")
+            return False
+
+        if not os.path.exists(control_program_path):
+            self._app.logger.error("Control Program file does not exist.")
+            return False
+
+        if not isinstance(control_program_args, str):
+            self._app.logger.error("Control Program arguments have to be a string.")
+            return False
+
+        self.props = copy.deepcopy(MaxwellTransient)
```

Comment:
Just for my knowledge, why do we need a deepcopy of MaxwellTransient

Reply:
@MaxJPRey honestly I saw we are already using the deepcopy() to copy other setups so I used it here as well and base on the explanation it constructs a new compound object and then, recursively, inserts copies into it of the objects found in the original. Rather than a shallow copy which inserts references into it to the objects found in the original.

But if you or anyone else have a better solution please do tell me :)

Reply:
This seems correct.